twogplus/views.py

```diff
@@ -60,17 +59,34 @@ def upload_cert():
         )
         return render_template("home.html")
 
-    # TODO If a user with the name already exist we should only update it
     username = vaccine_username if is_vaccinated else test_username
-    user = User(username, is_vaccinated, is_tested)
-    db.session.add(user)
-    db.session.commit()
-
-    # TODO set some cookies or shit
+    session["username"] = username
+
+    user_found = User.query.filter(User.name == username).first()
+    if user_found:
+        user = user_found
+        if is_vaccinated:
+            db.session.query(User).filter(User.name == username).update(
+                {"is_vaccinated": is_vaccinated}
+            )
+            db.session.commit()
+        if is_tested:
+            db.session.query(User).filter(User.name == username).update(
+                {"is_tested": is_tested}
+            )
+            db.session.commit()
+
+    else:
+        user = User(username, is_vaccinated, is_tested)
+        db.session.add(user)
+        db.session.commit()
 
     # TODO: propper message here
```

Comment:
This comment is now obsolete as the messages are now correct

Reply:
correct
```suggestion
```